packages/compass-indexes/src/components/create-index-form/query-flow-section.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  Button,
+  palette,
+  Body,
+  Code,
+  Link,
+} from '@mongodb-js/compass-components';
+import React from 'react';
+import { css, spacing } from '@mongodb-js/compass-components';
+import { CodemirrorInlineEditor } from '@mongodb-js/compass-editor';
+
+const inputQueryContainerStyles = css({
+  marginBottom: spacing[600],
+  border: `1px solid ${palette.gray.base}`,
+  borderRadius: spacing[300],
+  padding: spacing[600],
+  display: 'flex',
+  flexDirection: 'column',
+});
+
+const headerStyles = css({
+  marginBottom: spacing[200],
+});
+
+const suggestedIndexContainerStyles = css({
+  flexDirection: 'column',
+  display: 'flex',
+});
+
+const suggestedIndexButtonStyles = css({
+  float: 'right',
+  marginTop: spacing[400],
+});
+
+const programmingLanguageLinkStyles = css({
+  marginLeft: 'auto',
+  marginTop: spacing[100],
+});
+
+const QueryFlowSection = () => {
+  // TODO in CLOUDP-311786, replace hardcoded values with actual data
+  const db_name = 'sample_mflix';
+  const collection_name = 'comments';
+
+  const formatSuggestedIndex = () => {
+    return `
+db.getSiblingDB("${db_name}").getCollection("${collection_name}").createIndex(
+  "awards.win": "1",
+  "imdb.rating": "1",
+});
+`;
+  };
+
+  const [inputQuery, setInputQuery] = React.useState('');
+
+  return (
+    <>
+      <Body baseFontSize={16} weight="medium" className={headerStyles}>
+        Input Query
+      </Body>
+      <div className={inputQueryContainerStyles}>
+        <div>
+          <CodemirrorInlineEditor
+            data-testid="query-flow-section-json-editor"
+            language="json"
```

Comment:
We want the language here to be `language="javascript-expression"` so that it accepts shell syntax like
```ts
{
  _id: ObjectId("66db48fd73f134799d529645")
}
```
That's the default language of this property of the component so a change we can do is remove this line:
```suggestion
```

We should also remove the `initialJSONFoldAll` prop then too and update the `data-testid` to not mention json.